x-pack/solutions/security/plugins/elastic_assistant/server/lib/langchain/graphs/default_assistant_graph/index.ts

```diff
@@ -130,30 +130,31 @@ export const callAssistantGraph: AgentExecutor<true | false> = async ({
     }
   }
 
-  const agentRunnable = isOpenAI
-    ? await createOpenAIFunctionsAgent({
-        llm: createLlmInstance(),
-        tools,
-        prompt: formatPrompt(systemPrompts.openai, systemPrompt),
-        streamRunnable: isStream,
-      })
-    : llmType && ['bedrock', 'gemini'].includes(llmType)
-    ? await createToolCallingAgent({
-        llm: createLlmInstance(),
-        tools,
-        prompt:
-          llmType === 'bedrock'
-            ? formatPrompt(systemPrompts.bedrock, systemPrompt)
-            : formatPrompt(systemPrompts.gemini, systemPrompt),
-        streamRunnable: isStream,
-      })
-    : // used with OSS models
-      await createStructuredChatAgent({
-        llm: createLlmInstance(),
-        tools,
-        prompt: formatPromptStructured(systemPrompts.structuredChat, systemPrompt),
-        streamRunnable: isStream,
-      });
+  const agentRunnable =
+    isOpenAI || llmType === 'inference'
+      ? await createOpenAIToolsAgent({
```

Comment:
Note: I updated to `createOpenAIToolsAgent` since `createOpenAIFunctionsAgent` uses deprecated properties and the inference api expects the latest properties